src/main/java/rex/task/Task.java

```diff
@@ -0,0 +1,87 @@
+package rex.task;
+
+/**
+ * The {@code Task} class represents a generic task in the Rex application.
+ * It contains information about the task description, its completion status,
+ * and manages the count of tasks created.
+ */
+public class Task {
+    private static int numberOfTasks = 0;
+    private final String description;
+    private boolean isDone;
+
+    /**
+     * The {@code Task} class represents a generic task in the Rex application.
+     * It contains information about the task description, its completion status,
+     * and manages the count of tasks created.
+     */
+    protected Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+        numberOfTasks++;
+    }
+
+    /**
+     * Marks the task as done by setting its completion status to true.
+     */
+    public void markDone() {

```

Comment:
Would it be better if its markAsDone?

Reply:
I believe so, will make the necessary changes!